### PR TITLE
Homework3: Aleksandr Levdansky

### PR DESCRIPTION
**Pull request by Coromilly**
Third homework. Includes 3 tasks:
1) List practice;
2) Tuple practice;
3) Dictionary practice.

### CODE REVIEW

File: dictionary_practice.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+def count_characters(s):
+    """Function count and return the numbers of each character in a string."""
+    dct = {}
+    for symb in s:
+        if symb in dct:
+            dct[symb] += 1
+        else:
+            dct[symb] = 1
+    return {k: dct[k] for k in sorted(dct)}
```
Comment (belovmd) [inline dictionary_practice.py]: do we need this sort with high complexity and unclear result

**Review by belovmd — COMMENTED** [review]
Please put tasks descriptions.

File: dictionary_practice.py (hunk 1 above)
Comment (Coromilly) [reply dictionary_practice.py]: In your example function returns dictionary in alphabetical order. In my function return is the same) if this sort isn't needed i'll correct the program.

File: dictionary_practice.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    are square of keys. n – function argument. Default is 20.
+    """
+    return {num: num**2 for num in range(1, n + 1)}
+
+
+def count_characters(s):
+    """Function count and return the numbers of each character in a string."""
+    dct = {}
+    for symb in s:
+        if symb in dct:
```
Comment (belovmd) [inline dictionary_practice.py]: we have more generic way without if condition here - please take a look at dict methods

File: list_practice.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# 1. Use a list comprehension to construct the list
+# ['ab', 'ac', 'ad', 'bb', 'bc', 'bd'].
+lst1 = [x + y for x in ('a', 'b') for y in ('b', 'c', 'd')]
```
Comment (belovmd) [inline list_practice.py]: can we use 'ab' etc here?

Comment (belovmd) [inline list_practice.py]: can we use 'ab' etc here?

File: tuple_practice.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# 1. Create the list ['a', 'b', 'c'], then create a tuple from that list
+lst1 = list('abc')
```
Comment (belovmd) [inline tuple_practice.py]: do we need this type conversion here?

**Review by belovmd — COMMENTED** [review]
please address these issues

File: tuple_practice.py (hunk 4 above)
Comment (Coromilly) [reply tuple_practice.py]: the way of creating list isn't mentioned.

File: list_practice.py (hunk 3 above)
Comment (Coromilly) [reply list_practice.py]: solved.

Comment (Coromilly) [reply list_practice.py]: solved.

File: dictionary_practice.py (hunk 2 above)
Comment (Coromilly) [reply dictionary_practice.py]: solved.

**Review by belovmd — COMMENTED** [review]
8.5
it's maximal score for full completion of this hw.

**Pull request closed without merging**